src/main/java/com/microsoft/azure/relay/RelayedHttpListenerResponse.java

```diff
@@ -82,7 +83,8 @@ public RelayedHttpListenerContext getContext() {
 	 * RelayedHttpListenerResponse instance.
 	 */
 	public CompletableFuture<Void> closeAsync() {
-		return this.outputStream.closeAsync().whenComplete(($void, ex) -> this.disposed = true);
+		CompletableFuture<Void> closeTask = (this.outputStream != null) ? this.outputStream.closeAsync() : CompletableFuture.completedFuture(null);
```

Comment:
>CompletableFuture<Void> closeTask = (this.outputStream != null) ? this.outputStream.closeAsync() : CompletableFuture.completedFuture(null); [](start = 2, length = 139)

This is personal preference but for me this pretty much crosses my threshold to use an actual 'if' statement.

Just saying, your choice to "Won't Fix" if you want.